modules/ldap-syncing-spec.adoc

```diff
@@ -0,0 +1,250 @@
+if not set.
+|string
+
+|`derefAliases`
+|The (optional) behavior of the search with regards to alisases. Can be `never`
+(never dereference aliases), `search` (only dereference in searching), `base`
+(only dereference in finding the base object), `always` (always dereference).
+Defaults to `always` if not set.
+|string
+
+|`timeout`
+|Holds the limit of time in seconds that any request to the server can remain outstanding before the wait for a response is given up. If this is `0`, no client-side limit is imposed.
+|integer
+
+|`filter`
+|A valid LDAP search filter that retrieves all relevant entries from the LDAP server with the base DN.
+|string
+
+|`pageSize`
+|Maximum preferred page size, measured in LDAP entries. A page size of `0` means no paging will be done.
+|integer
+|===
+
+[[sync-ldap-v1-rfc2307config]]
+== v1.RFC2307Config
+
+`RFC2307Config` holds the necessary configuration options to define how an LDAP
+group sync interacts with an LDAP server using the RFC2307 schema.
+
+[options="header"]
+|===
+|Name |Description |Schema
+
+|`groupsQuery`
+|Holds the template for an LDAP query that returns group entries.
+|v1.LDAPQuery
+
+|`groupUIDAttribute`
+|Defines which attribute on an LDAP group entry will be interpreted as its unique identifier. (`ldapGroupUID`)
+|string
+
+|`groupNameAttributes`
+|Defines which attributes on an LDAP group entry will be interpreted as its name to use for an {product-title} group.
+|string array
+
+|`groupMembershipAttributes`
+|Defines which attributes on an LDAP group entry will be interpreted as its members. The values contained in those attributes must be queryable by your `UserUIDAttribute`.
+|string array
+
+|`usersQuery`
+|Holds the template for an LDAP query that returns user entries.
+|v1.LDAPQuery
+
+|`userUIDAttribute`
+|Defines which attribute on an LDAP user entry will be interpreted as its unique identifier. It must correspond to values that will be found from the `GroupMembershipAttributes`.
+|string
+
+|`userNameAttributes`
+|Defines which attributes on an LDAP user entry will be used, in order, as its {product-title} user name. The first attribute with a non-empty value is used. This should match your `PreferredUsername` setting for your `LDAPPasswordIdentityProvider`. The attribute to use as the name of the user in the {product-title} group
+record. `mail` or `sAMAccountName` are preferred choices in most installations.
+|string array
+
+|`tolerateMemberNotFoundErrors`
+|Determines the behavior of the LDAP sync job when missing user entries are encountered. If `true`, an LDAP query for users that does not find any will be tolerated and an only and error will be logged. If `false`, the LDAP sync job will fail if a query for users doesn't find any. The default value is 'false'. Misconfigured LDAP sync jobs with this flag set to 'true' can cause group membership to be removed, so it is recommended to use this flag with caution.
```

Comment:
I'd put true and false in backticks throughout this description, instead of switching to single quotes.